scipy/stats/_bootstrap.py

```diff
@@ -0,0 +1,375 @@
+        percentiles[indices] = np.percentile(theta_hat_b_i, alpha_i)
+    return percentiles[()]  # return scalar instead of 0d array
+
+
+def _bca_interval(data, statistic, axis, alpha, theta_hat_b):
+    """Bias-corrected and accelerated interval."""
+    # closely follows [2] "BCa Bootstrap CIs"
+    sample = data[0]  # only works with 1 sample statistics right now
+
+    # calculate z0_hat
+    theta_hat = statistic(sample, axis=axis)[..., None]
+    percentile = _percentile_of_score(theta_hat_b, theta_hat, axis=-1)
+    z0_hat = ndtri(percentile)
+
+    # calculate a_hat
+    jackknife_sample = _jackknife_resample(sample)
+    theta_hat_i = statistic(jackknife_sample, axis=-1)
+    theta_hat_dot = theta_hat_i.mean(axis=-1, keepdims=True)
+    num = ((theta_hat_dot - theta_hat_i)**3).sum(axis=-1)
+    den = 6*((theta_hat_dot - theta_hat_i)**2).sum(axis=-1)**(3/2)
+    a_hat = num / den
+
+    # calculate alpha_1, alpha_2
+    z_alpha = ndtri(alpha)
+    z_1alpha = -z_alpha
+    num1 = z0_hat + z_alpha
+    alpha_1 = ndtr(z0_hat + num1/(1 - a_hat*num1))
+    num2 = z0_hat + z_1alpha
+    alpha_2 = ndtr(z0_hat + num2/(1 - a_hat*num2))
+    return alpha_1, alpha_2
+
+
+def _bootstrap_ci_iv(data, statistic, axis, confidence_level, n_resamples,
+                     method, random_state):
+    """Input validation for `bootstrap_ci`."""
+    axis_int = int(axis)
+    if axis != axis_int:
+        raise ValueError("`axis` must be an integer.")
+
+    n_samples = 0
+    try:
+        n_samples = len(data)
+    except TypeError:
+        raise ValueError("`data` must be a sequence of samples.")
+
+    if n_samples == 0:
+        raise ValueError("`data` must contain at least one sample.")
+
+    data_iv = []
+    for sample in data:
+        sample = np.atleast_1d(sample)
+        if sample.shape[axis] <= 1:
+            raise ValueError("each sample in `data` must contain two or more "
+                             "observations along `axis`.")
+        sample = np.moveaxis(sample, axis_int, -1)
+        data_iv.append(sample)
+
+    # should we try: statistic(data, axis=axis) here?
+
+    confidence_level_float = float(confidence_level)
+
+    n_resamples_int = int(n_resamples)
+    if n_resamples != n_resamples_int or n_resamples_int <= 0:
+        raise ValueError("`n_resamples` must be a positive integer.")
+
+    methods = {'percentile', 'basic', 'bca'}
+    method = method.lower()
+    if method not in methods:
+        raise ValueError(f"`method` must be in {methods}")
+
+    message = "`method = 'BCa' is only available for one-sample statistics"
+    if n_samples > 1 and method == 'bca':
+        raise ValueError(message)
+
+    random_state = check_random_state(random_state)
+
+    return (data_iv, statistic, axis_int, confidence_level_float,
+            n_resamples_int, method, random_state)
+
+
+def bootstrap_ci(data, statistic, axis=0, confidence_level=0.95,
+                 n_resamples=1000, method='basic', random_state=None):
```

Comment:
We might consider using a larger default for `n_resamples`.  The paper

>    Tim Hesterberg (2015), What Teachers Should Know about the Bootstrap: Resampling in the Undergraduate Statistics Curriculum, The American Statistician 69(4) 371-386, DOI: 10.1080/00031305.2015.1089789

includes a discussion of how many samples to use, and recommends 10000.  (A longer version of the paper is at https://arxiv.org/abs/1411.5279.)

Also, in his response to @tupui, Art Owen recommended 9999.

Reply:
I am just worried of the performance impact here. Maybe this would be too much? Or we might want to parallelise the calls.

Reply:
I'm worried (well, maybe "worried" is too strong) that we will spend the next 10 years telling users something like "The default of 1000 was based on technology available in the '80s; these days no one uses less than 10000".

Reply:
I actually did not do any check here so my fear is probably not that founded :) Fine for me to go with 9999. After all it's to have confidence intervals and this is known to take ressources.

Reply:
The rule-of-thumb is 250 to 500 resamples to estimate a bootstrap standard error and at least 10,000 for a confidence interval. On a modern computer you can do 50,000 or 100,000 with impunity, at least if you use C or Fortran.

Reply:
Sounds like it's between 9999 and 10000 for the default. Would you mind casting a vote for one of the two @sturlamolden @tupui @WarrenWeckesser?

Reply:
9999 on my side to follow Art's comment.

Reply:
No preference.  I don't think the code will take advantage of 9999 being a  "nice round number minus 1", so I don't think it matters.

Reply:
Out of curiosity, what was Art’s argument for preferring an odd number? 

Personally I have no preference between 9,999 and 10,000, those are practially the same as far as I am concerned, so I have no opinion on that.

Reply:
@sturlamolden it's that for a segment cut in 9999 you would have 10000 sections. See here https://github.com/scipy/scipy/pull/13371#issuecomment-831341953.